Error: `pipeline::ShaderError<E>` doesn't implement `std::fmt::Display`

<!-- Thank you for filing this! Please read the [debugging tips](https://github.com/gfx-rs/wgpu/wiki/Debugging-wgpu-Applications).
That may let you investigate on your own, or provide additional information that helps us to assist.-->

**Description**
Just updating a project to use `wgpu` latest master, getting a validation error.

```
error[E0277]: `pipeline::ShaderError<E>` doesn't implement `std::fmt::Display`
  --> /home/kvark/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/2755cd0/wgpu-core/src/pipeline.rs:50:1
   |
50 | pub struct ShaderError<E> {
   | ^^^^^^^^^^^^^^^^^^^^^^ `pipeline::ShaderError<E>` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `pipeline::ShaderError<E>`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
help: consider introducing a `where` bound, but there might be an alternative better way to express this requirement
   |
50 | pub struct ShaderError<E> where pipeline::ShaderError<E>: std::fmt::Display {
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: `E` doesn't implement `std::fmt::Debug`
  --> /home/kvark/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/2755cd0/wgpu-core/src/pipeline.rs:50:1
   |
50 | pub struct ShaderError<E> {
   | ^^^^^^^^^^^^^^^^^^^^^^ `E` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
   |
note: required because of the requirements on the impl of `std::fmt::Debug` for `pipeline::ShaderError<E>`
  --> /home/kvark/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/2755cd0/wgpu-core/src/pipeline.rs:49:17
   |
49 | #[derive(Clone, Debug, Error)]
   |                 ^^^^^
   = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `E`
   |
50 | pub struct ShaderError<E: std::fmt::Debug> {
```

**Repro steps**

**Expected vs observed behavior**
Has to compile and work.

**Extra materials**
This is from https://github.com/kvark/vange-rs

**Platform**
Linux

--- a/wgpu-core/src/pipeline.rs
+++ b/wgpu-core/src/pipeline.rs
@@ -5,7 +5,7 @@ use crate::{
     id::{DeviceId, PipelineLayoutId, ShaderModuleId},
     validation, Label, LifeGuard, Stored,
 };
-use std::{borrow::Cow, fmt};
+use std::{borrow::Cow, error::Error, fmt};
 use thiserror::Error;
 
 #[allow(clippy::large_enum_variant)]
@@ -46,11 +46,10 @@ impl<A: hal::Api> Resource for ShaderModule<A> {
     }
 }
 
-#[derive(Clone, Debug, Error)]
+#[derive(Clone, Debug)]
 pub struct ShaderError<E> {
     pub source: String,
     pub label: Option<String>,
-    #[source]
     pub inner: E,
 }
 impl fmt::Display for ShaderError<naga::front::wgsl::ParseError> {
@@ -89,6 +88,15 @@ impl fmt::Display for ShaderError<naga::WithSpan<naga::valid::ValidationError>> 
             "\nShader validation {}",
             String::from_utf8_lossy(&writer.into_inner())
         )
+    }
+}
+impl<E> Error for ShaderError<E>
+where
+    ShaderError<E>: fmt::Display,
+    E: Error + 'static,
+{
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        Some(&self.inner)
     }
 }
 

